src/DependencyInjection/MercureExtension.php

```diff
@@ -67,9 +73,29 @@ public function load(array $configs, ContainerBuilder $container)
             $bus = $hub['bus'] ?? null;
             $attributes = null === $bus ? [] : ['bus' => $hub['bus']];
             $publisherDefinition->addTag('messenger.message_handler', $attributes);
+
+            if ($config['enable_profiler'] && class_exists(Stopwatch::class)) {
+                $container->register("$hubId.traceable", TraceablePublisher::class)
+                    ->setDecoratedService($hubId)
+                    ->setArguments([new Reference("$hubId.traceable.inner"), new Reference('debug.stopwatch')]);
+                $publishers[$name] = new Reference("$hubId.traceable");
+                if ($defaultHub === $hubId) {
+                    $container->setAlias(TraceablePublisher::class, "$hubId.traceable");
+                }
+            }
+        }
+
+        if ($config['enable_profiler']) {
+            $container->register('data_collector.mercure', MercureDataCollector::class)
```

Comment:
This definition should be in the XML file and then removed here if `$config['enable_profiler']` is false. Doing this allows to have autocompletion in PHPStorm and the like.

Reply:
To do so, I'll need to tag the publisher services and handle them through a CompilerPass to inject them in the MercureDataCollector...

Isn't there an easier way to handle that?